📝 docs: Add oks_manifest resource (#755)
outscale/terraform-provider-outscale: docs/resources/oks_manifest.md

### Required resources

```hcl
resource "outscale_oks_project" "project" {
  name            = "project01"
  cidr            = "10.50.0.0/18"
  region          = "eu-west-2"
}

resource "outscale_oks_cluster" "cluster" {
  project_id      = outscale_oks_project.project.id
  admin_whitelist = ["0.0.0.0/0"]
  cidr_pods       = "10.91.0.0/16"
  cidr_service    = "10.92.0.0/16"
  version         = "1.35"
  name            = "cluster01"
  control_planes  = "cp.mono.master"
}
```

### Create an OKS node pool

```hcl
resource "outscale_oks_manifest" "nodepool" {
  cluster_id = outscale_oks_cluster.cluster.id
  manifest   = <<-YAML
apiVersion: oks.dev/v1beta2
kind: NodePool
metadata:
  name: pool-1
spec:
  autoHealing: true
  desiredNodes: 1
  nodeType: tinav7.c1r1p1
  upgradeStrategy:
    autoUpgradeEnabled: true
    autoUpgradeMaintenance:
      durationHours: 1
      startHour: 12
      weekDay: Tue
    maxSurge: 0
    maxUnavailable: 1
  volumes:
  - device: root
    dir: /
    size: 100
    type: gp2
  zones: [eu-west-2a]
YAML

  wait_for = {
    fields = {
      "status.progress.ready" = "1"
    }
  }
}
```

## Argument Reference

The following arguments are supported:

> * `cluster_id` - (Required) The cluster ID on which the manifest is applied.

Reviewer: ```suggestion
* `cluster_id` - (Required) The ID of the cluster on which you want to apply the manifest.
```

* `manifest` - (Required) The Kubernetes YAML manifest.

> * `skip_delete` (Optional, default: false) If set to true, Terraform removes only the manifest resource from the state during delete, without deleting the Kubernetes object from the cluster.

Reviewer: ```suggestion
* `skip_delete` (Optional) If set to true, Terraform removes only the manifest resource from the state during delete, without deleting the Kubernetes object from the cluster. By default, false.
```

> * `wait` - (Optional, default: false): If set to true, Terraform waits for the Kubernetes object to be deleted during destroy.

Reviewer: ```suggestion
* `wait` - (Optional): If set to true, Terraform waits for the Kubernetes object to be deleted during destroy. By default, false.
```

* `wait_for` - (Optional) Wait until the fields in the Kubernetes object match the expected values after apply:

>     * `fields` - (Required) Maps of key-value pairs (field path => expected pattern).

Reviewer: ```suggestion
    * `fields` - (Required) Maps of key/value pairs (field path => expected pattern).
```

        * Each key must be a [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) field path, but a simple field path such as `status.progress.ready` is accepted and converted internally to a JSONPath expression (that is, the enclosing `{` `}` and the first `.` can be omitted).
        * Each value is a regex pattern.
        * All configured fields must match for the wait to complete.

>         * Examples: `"status.progress.ready" = "1"`, `"{.status.progress.ready}" = "1"`, `"{.status.state.name}" = "idle|reconciliation"`.

Reviewer: ```suggestion
    * `fields` - (Required) Maps of key/value pairs in the following format: `"{.field.path}" = "expected_pattern"`.<br />
Each key must be a [JSONPath expression](https://kubernetes.io/docs/reference/kubectl/jsonpath/), where the enclosing characters (`{` `}` and the first `.`) are optional, and each value must be a regex pattern. All configured fields must match for the wait to complete.<br />
Examples: `"status.progress.ready" = "1"`, `"{.status.progress.ready}" = "1"`, `"{.status.state.name}" = "idle|reconciliation"`.
```

Author: I applied a slightly modified version of this

>     * `timeout` - (Optional) Custom timeout for the `wait_for` checks. If not specified, falls back to the CRUD operation default timeout.

Reviewer: ```suggestion
    * `timeout` - (Optional) A custom timeout for the `wait_for` checks. If not specified, falls back to the CRUD operation default timeout.
```
